### PR TITLE
dont defer init for assumepageview

### PR DESCRIPTION
**Pull request by hankim813**
https://segment.atlassian.net/browse/EPD-798

# TL:DR
- issue: ready function doesn't fire until you call page if you have integrations enabled with assumepageview option. so problematic if you have `.page()` call inside `.ready()`
- blocker for lendingtree implementing plugin code for adobe since it needs to be in `.ready()`

**old behavior**:

1) defer init to the 2nd time you call `.initialize()
2) wrap `.page()` call to .initialize() if `.assumePageview()` is true and you haven't yet initialized
3) effectively treat first `.page()` call to be for initializing

**new behavior**:

1) initialize all integrations upon loading analytics.js
2) if you have `.assumePageview()`, noop the first page call and mark an internal metadata keeping track of this state
3) document for [all integrations](https://github.com/search?p=6&q=org%3Asegment-integrations+assumespageview&ref=searchresults&type=Code&utf8=%E2%9C%93) this caveat that the partner library invokes its own internal pageview event upon loading, NOT when you first call `.page()` call

here is a longer brief for this issue in detail that sperandio wrote here: https://paper.dropbox.com/doc/Research-AssumesPageview-Fix-dpTYzw1pW9infEElyWNvS

@sperand-io @jgershen @tsholmes @ndhoule @f2prateek 

any objections?

### CODE REVIEW

**Comment by jgershen** [comment]
I like this / agree with the new behavior at a high level! Defer to @ndhoule @tsholmes for a real code review.

File: lib/protos.js [hunk 1]
```diff
@@ -350,8 +342,10 @@ exports._wrapInitialize = function() {
 exports._wrapPage = function() {
   var page = this.page;
   this.page = function() {
-    if (this._assumesPageview && !this._initialized) {
-      return this.initialize.apply(this, arguments);
+    if (this._assumesPageview) {
```
Comment (tsholmes) [inline lib/protos.js]: We could just have this be `this.page = after(2, this.page)` when `assumesPageView` is set, which will just noop the first call. Then we don't have to keep track of any additional state.

Comment (hankim813) [inline lib/protos.js]: genius!!
